### DISCUSSION TITLE
Element wise exponentiation is implemented in OpenModelica?

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#11274] [Q&A]
Hi,

I wonder if OpenModelica implements the element wise exponentiation of arrays, since I get error in the compiler when I try to use:
``` Modelica
function F 
  input Real c[3];
  input Real x[3];
  output Real y[3];
  algorithm
    y := x.^c;
end F;
```

but it works fine when use:
`y := x.^2` instead of `y := x.^c`

The error is:
clang -municode  -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse     -I"C:/Program Files/OpenModelica1.23.0-dev-64bit/include/omc/c" -I"C:/Program Files/OpenModelica1.23.0-dev-64bit/include/omc" -I. -DOPENMODELICA_XML_FROM_FILE_AT_RUNTIME -DOMC_MODEL_PREFIX=drF2cmpLivGut -DOMC_NUM_MIXED_SYSTEMS=0 -DOMC_NUM_LINEAR_SYSTEMS=0 -DOMC_NUM_NONLINEAR_SYSTEMS=0 -DOMC_NDELAY_EXPRESSIONS=3 -DOMC_NVAR_STRING=0  -c -o drF2cmpLivGut_13opt.o drF2cmpLivGut_13opt.c
drF2cmpLivGut_functions.c:326:24: error: use of undeclared identifier 'daeExpBinary'
  real_array_copy_data(daeExpBinary:ERR for POW_ARR2, _y);
                       ^
Thank you in advance!

**Comment by perost** [comment]
Yes, all the arithmetic operators should work. I can't reproduce the issue using only the function you provided, so it might be that the compiler does something with the function based on how you use it in your model that goes wrong.

It would be helpful if you could share the full model if that's possible. You can also open an issue for it instead of continuing here so we can track the issue properly, since it's definitely an issue with the compiler and not your model.

**Reply by mpradovelasco** [reply]
OK, I have opened the issue #11275 that shows a MWE and present this problem and one additional related to it. 

Thanks
